ambry-clustermap/src/main/java/com.github.ambry.clustermap/Partition.java

```diff
@@ -44,7 +44,7 @@
   private static final short Current_Version = 1;
   private static final int Partition_Size_In_Bytes = Version_Field_Size_In_Bytes + 8;
 
-  private final List<Replica> replicas;
+  private final List<ReplicaId> replicas;
```

Comment:
Any reason to change `Replica` to `ReplicaId`?  `Partittion`, `Replica` and `Disk` are for static clustermap. I think in this class, it is supposed to be `Replica` and cannot be another implementation of `ReplicaId`.

Reply:
As discussed offline this was needed to make APIs generic, so that merging on cluster maps becomes easy in RecoveryTestClusterManager.